ethereum/blockcreation/src/main/java/org/hyperledger/besu/ethereum/blockcreation/AbstractMiningCoordinator.java

```diff
@@ -63,75 +72,116 @@ public AbstractMiningCoordinator(
       final BlockHeader parentHeader,
       final List<Transaction> transactions,
       final List<BlockHeader> ommers) {
-    M miner = executor.createMiner(parentHeader);
+    final M miner = executor.createMiner(Subscribers.none(), parentHeader);
     return Optional.of(miner.createBlock(parentHeader, transactions, ommers));
   }
 
   @Override
-  public void enable() {
+  public void start() {
     synchronized (this) {
-      if (isEnabled) {
+      if (state != State.IDLE) {
+        return;
+      }
+      state = State.RUNNING;
+      startMiningIfPossible();
+    }
+  }
+
+  @Override
+  public void stop() {
+    synchronized (this) {
+      if (state != State.RUNNING) {
         return;
       }
-      if (syncState.isInSync()) {
-        startAsyncMiningOperation();
+      state = State.STOPPED;
+      haltCurrentMiningOperation();
+      executor.shutDown();
+    }
+  }
+
+  @Override
+  public void awaitStop() throws InterruptedException {
+    executor.awaitShutdown();
+  }
+
+  @Override
+  public boolean enable() {
+    synchronized (this) {
+      if (isEnabled) {
+        return true;
       }
       isEnabled = true;
+      startMiningIfPossible();
     }
+    return true;
   }
 
   @Override
-  public void disable() {
+  public boolean disable() {
     synchronized (this) {
       if (!isEnabled) {
-        return;
+        return false;
       }
-      haltCurrentMiningOperation();
       isEnabled = false;
+      haltCurrentMiningOperation();
     }
+    return false;
   }
 
   @Override
-  public boolean isRunning() {
+  public boolean isMining() {
     synchronized (this) {
       return currentRunningMiner.isPresent();
     }
   }
 
-  protected void startAsyncMiningOperation() {
+  private synchronized boolean startMiningIfPossible() {
```

Comment:
nit: not obvious what the return value is - maybe don't return anything - then people can query the currentMiner to see if mining is running? (only in the SyncChanged function)